lib/storage/bucket.js

```diff
@@ -203,6 +203,100 @@ function Bucket(storage, name) {
 }
 
 /**
+ * Combine mutliple files into one new file.
+ *
+ * @throws if a non-array is provided as sources argument.
+ * @throws if less than two sources are provided.
+ * @throws if no destination is provided.
+ * @throws if a content type cannot be determined for the destination file.
+ *
+ * @param {string[]|module:storage/file[]} sources - The source files that will
+ *     be combined.
+ * @param {string|module:storage/file} destination - The file you would like the
+ *     source files combined into.
+ * @param {function=} callback - The callback function.
+ *
+ * @example
+ * var 2013logs = bucket.file('2013-logs.txt');
+ * var 2014logs = bucket.file('2014-logs.txt');
+ *
+ * var allLogs = bucket.file('all-logs.txt');
+ *
+ * bucket.combine([
+ *   2013logs,
+ *   2014logs
+ * ], allLogs, function(err, newFile) {
+ *   // newFile === allLogs
+ * });
+ */
+Bucket.prototype.combine = function(sources, destination, callback) {
+  if (!util.is(sources, 'array') || sources.length < 2) {
+    throw new Error('You must provide at least two source files.');
+  }
+
+  if (!destination) {
+    throw new Error('A destination file must be specified.');
+  }
+
+  var self = this;
```

Comment:
Woops! This was actually a mistake. I hate `that = this`, too, but would have done it to match the codebase. I'll be so happy to keep this and work towards removing other that's.